lib/gooddata/lcm/actions/synchronize_users.rb

```diff
@@ -262,7 +263,13 @@ def call(params)
                         fail "Client id cannot be empty" if client_id.blank?
 
                         c = domain_clients.detect { |specific_client| specific_client.id == client_id }
-                        fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\"" if c.nil?
+                        if c.nil?
+                          filtered_client = all_domain_clients.detect { |f_client| f_client.id == client_id }
+                          fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\"" if filtered_client.nil?
+
+                          GoodData.logger.info("Client #{client_id} is outside specific clients in segments_filter #{segment_uris}")
```

Comment:
please update the message:
"Client \"#{client_id}\" is not belong to filtered segments"